test/query_dialog_test.rb

```diff
@@ -0,0 +1,90 @@
+#! rspec
+# Copyright (c) 2014 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact Novell about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require_relative "spec_helper"
+require "systemd_journal/query_dialog"
+
+describe SystemdJournal::QueryDialog do
+  def property(id, property)
+    Yast::UI.QueryWidget(Id(id), property)
+  end
+
+  def input(value, after_doing: nil)
```

Comment:
This took me a moment to realize so I think it worth commenting:
> \# We need to tie expectations to input events
\# because after the whole dialog has run it will be closed and we cannot query widgets anymore.

"after_doing" sounds slightly unusual. Maybe "post_hook"?